From 82f7f3de0e90bb7d1447cb710ffd2e1b315ae479 Mon Sep 17 00:00:00 2001
From: Artem Dyomin <artem.dyomin@qt.io>
Date: Fri, 26 May 2023 15:15:52 +0200
Subject: [PATCH] Remove std::unary_function usage

std::unary_function was removed with c++17

EDITED BY @cbjeukendrup to adapt for Qt 6.2.9

Pick-to: 6.5
Task-number: QTBUG-113782
Change-Id: I4e330cd1f89dc14936acbccdeee8378ea4938870
Reviewed-by: Lars Knoll <lars@knoll.priv.no>
Reviewed-by: Qt CI Bot <qt_ci_bot@qt-project.org>
Reviewed-by: Pavel Dubsky <pavel.dubsky@qt.io>
---

diff --git a/src/multimedia/platform/darwin/camera/avfcamerautility.mm b/src/multimedia/platform/darwin/camera/avfcamerautility.mm
index 444162523..513654618 100644
--- a/src/multimedia/platform/darwin/camera/avfcamerautility.mm
+++ b/src/multimedia/platform/darwin/camera/avfcamerautility.mm
@@ -102,9 +102,9 @@ struct ByResolution
     }
 };
 
-struct FormatHasNoFPSRange : std::unary_function<AVCaptureDeviceFormat *, bool>
+struct FormatHasNoFPSRange
 {
-    bool operator() (AVCaptureDeviceFormat *format)
+    bool operator() (AVCaptureDeviceFormat *format) const
     {
         Q_ASSERT(format);
         return !format.videoSupportedFrameRateRanges || !format.videoSupportedFrameRateRanges.count;
